source/common/http/filter_manager.cc

```diff
@@ -867,6 +867,10 @@ FilterManager::commonDecodePrefix(ActiveStreamDecoderFilter* filter,
 }
 
 void DownstreamFilterManager::onLocalReply(StreamFilterBase::LocalReplyData& data) {
+  // To ensure we have filters over which we can iterate and call onLocalReply.
+  // If the filter chain already exists this will be a no-op.
+  createFilterChain();
```

Comment:
Why we need to add this? This is too generic place for fixing an ext_proc filter issue.

Reply:
I'm pulling this out into a separate PR. On protocol errors the filters do not exist when this is called so none of the filters' onLocalReply callbacks are run

Reply:
This change has been removed from this PR and addressed separately in https://github.com/envoyproxy/envoy/pull/29501

Reply:
https://github.com/envoyproxy/envoy/pull/29501 is the base of this PR. You probably want to get that PR works first.

Reply:
That is why this is a draft. I am blocked on #29501 on mobile issues for which I have very little context but which appear to rely on the bug.